hashintel/hash
# Introduce ability to create block-level comments

## 🌟 What is the purpose of this PR?

This PR adds the logic for creating Comments attached to a block

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1201095311341924/1202882932767484/f) _(internal)_
- [PRD](https://www.notion.so/hashintel/PRD-Comments-v1-6148130ae28f461f96b9134e1ed2f1b3)
- [Development  Doc](https://www.notion.so/hashintel/Comments-Development-b805a75ac7d448feae01517c09ca2ccd)
- [Figma](https://www.figma.com/file/gydVGka9FjNEg9E2STwhi2/HASH-App?node-id=182%3A5795)

## 🔍 What does this change?

- Added Comment as System type;
- Added logic to create comments in the API;
- Added a comment button to the right of each block that enables users to create comments attached to that block;
- Made several changes to the prosemirror plugins and schema so that they can be reused in the comment input prosemirror instance too.

## 🐾 Next steps

Next tasks in the [Milestone](https://app.asana.com/0/1200211978612931/1202689833684554/f)


## ❓ How to test this?

1. Run yarn seed-data;
2. Open the Comment input by clicking on the button on the right of a block;
3. Type on the input, test formatting (cmd + b, cmd + i, cmd + u) and mentions (@};
4. Press enter to create comment.

The comment should be created but there's no visual feedback on the app yet, the only way to check it is by checking the db.

## 📹 Demo

![comment](https://user-images.githubusercontent.com/37453800/189933958-c5acf337-562d-4d77-bc19-9ab209566a1d.gif)
